File: app/src/main/java/wangdaye/com/geometricweather/db/entities/LocationEntity.java
package wangdaye.com.geometricweather.db.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.annotation.NonNull;
import java.util.TimeZone;
import wangdaye.com.geometricweather.common.basic.models.options.provider.WeatherSource;

/**
 * Location entity matching GreenDAO schema 62.
 */
@Entity(tableName = "LOCATION_ENTITY")
public class LocationEntity {

    @PrimaryKey
    @ColumnInfo(name = "FORMATTED_ID")
    @NonNull
    public String formattedId = "";

    @ColumnInfo(name = "CITY_ID")
    public String cityId;

    @ColumnInfo(name = "LATITUDE")
    public float latitude;

    @ColumnInfo(name = "LONGITUDE")
    public float longitude;

    @ColumnInfo(name = "TIME_ZONE")
    public TimeZone timeZone;

    @ColumnInfo(name = "COUNTRY")
    public String country;

    @ColumnInfo(name = "PROVINCE")
    public String province;

    @ColumnInfo(name = "CITY")
    public String city;

    @ColumnInfo(name = "DISTRICT")
    public String district;

    @ColumnInfo(name = "WEATHER_SOURCE")
    public WeatherSource weatherSource;

    @ColumnInfo(name = "CURRENT_POSITION")
    public boolean currentPosition;

    @ColumnInfo(name = "RESIDENT_POSITION")
    public boolean residentPosition;

    @ColumnInfo(name = "CHINA")
    public boolean china;

    public LocationEntity() {
    }
}

File: app/src/main/java/wangdaye/com/geometricweather/db/entities/LocationEntity.kt
package wangdaye.com.geometricweather.db.entities

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.TimeZone
import kotlin.jvm.JvmField
import wangdaye.com.geometricweather.common.basic.models.options.provider.WeatherSource

/**
 * Location entity matching GreenDAO schema 62.
 */
@Entity(tableName = "LOCATION_ENTITY")
class LocationEntity {

    @PrimaryKey
    @ColumnInfo(name = "FORMATTED_ID")
    @NonNull
    @JvmField
    var formattedId: String = ""

    @ColumnInfo(name = "CITY_ID")
    @JvmField
    var cityId: String? = null

    @ColumnInfo(name = "LATITUDE")
    @JvmField
    var latitude: Float = 0f

    @ColumnInfo(name = "LONGITUDE")
    @JvmField
    var longitude: Float = 0f

    @ColumnInfo(name = "TIME_ZONE")
    @JvmField
    var timeZone: TimeZone? = null

    @ColumnInfo(name = "COUNTRY")
    @JvmField
    var country: String? = null

    @ColumnInfo(name = "PROVINCE")
    @JvmField
    var province: String? = null

    @ColumnInfo(name = "CITY")
    @JvmField
    var city: String? = null

    @ColumnInfo(name = "DISTRICT")
    @JvmField
    var district: String? = null

    @ColumnInfo(name = "WEATHER_SOURCE")
    @JvmField
    var weatherSource: WeatherSource? = null

    @ColumnInfo(name = "CURRENT_POSITION")
    @JvmField
    var currentPosition: Boolean = false

    @ColumnInfo(name = "RESIDENT_POSITION")
    @JvmField
    var residentPosition: Boolean = false

    @ColumnInfo(name = "CHINA")
    @JvmField
    var china: Boolean = false
}
